### PR TITLE
FIX Use correct host on response destination fallback check

### LINKED ISSUE #151: Destination check contains unnecessary duplicate check
see: https://github.com/onelogin/php-saml/blob/570a5d9cc42a1a205355b9212bd182fbe35404d9/lib/Saml2/Response.php#L170-L181

If the check for whether the destination starts with current url fails, destination is checked with the `OneLogin_Saml2_Utils::getSelfRoutedURLNoQuery()` - however this is the initial value of  the first check (see https://github.com/onelogin/php-saml/blob/570a5d9cc42a1a205355b9212bd182fbe35404d9/lib/Saml2/Response.php#L133)

The repeated check does not appear to be needed.

### PR DESCRIPTION
**Pull request by dhensby**
Fixes #151 

I'm not sure how I can test this in the current test suite - any advice?

### CODE REVIEW

File: lib/Saml2/Response.php [hunk 1]
```diff
@@ -171,7 +171,7 @@ public function isValid($requestId = null)
                     $destination = trim($this->document->documentElement->getAttribute('Destination'));
                     if (!empty($destination)) {
                         if (strpos($destination, $currentURL) !== 0) {
-                            $currentURLrouted = OneLogin_Saml2_Utils::getSelfRoutedURLNoQuery();
+                            $currentURLrouted = OneLogin_Saml2_Utils::getSelfURLNoQuery();
```
Comment (pitbulk) [inline lib/Saml2/Response.php]: can you rename the $currentURLrouted by $currentURLNoRouted ?

Comment (dhensby) [inline lib/Saml2/Response.php]: sure

Comment (dhensby) [inline lib/Saml2/Response.php]: done

**Comment by pitbulk** [comment]
I plan to add some improvements on the toolkit in the next weeks, so I will take care of the test of that part.

**Comment by dhensby** [comment]
Thanks

**Pull request merged**